### PR TITLE
add switch statement conventions to style guide

### LINKED ISSUE #75: Style guide: Add section about switch
See comment on #73 re: switch and add that to the style guide.

### PR DESCRIPTION
**Pull request by ericvw**
I know this is going to be a contentious discussion.

Closes #75

### CODE REVIEW

**Comment by ericlu88** [comment]
LGTM.

**Comment by yupinghu** [comment]
I guess I'll have to mess with my editor config... LGTM. @thmiceli ?

**Comment by thmiceli** [comment]
I am a little annoyed by the 2-space indent when everything else is 4 spaces. That will be a problem with editors that do automatic indentation. If this is to save horizontal space I'd rather align the 'case' tags with the 'switch'.

**Comment by ericvw** [comment]
Part of it is to save horizontal space, but the primary motivation is that the code is still indented by 4-spaces from the top-level `switch` keyword.  This indentation rule is used throughout the [BDE](http://github.com/bloomberg.com/bde) project for `switch` statements, and I proposed this style to keep consistent with existing Bloomberg conventions.

However, I am open to tweaking the rules on this (hence this PR) :).

**Comment by ericvw** [comment]
Updated to address the no-op `default` case bodies.  Still investigating text editor behavior/configuration regarding the 2-space, 2-space style currently presented.

`vim` and `emacs` can be configured to handle this style.  Researching into others.

**Comment by ericvw** [comment]
Revised `no-op` conventions.

**Comment by ericvw** [comment]
The 2/4 indentation has been decided - ready for final sign-off's and merge.

File: doc/style-guide.md [hunk 1]
```diff
@@ -92,6 +96,35 @@ quaility.
...
+  } break;
+  default: {
+    /* always required */
+  } break;
+}
+```
+
+  - If a case statement is an intentional no-op, then a comment, `// no-op`, should be within the
+    respective case body to indicate the intended behavior.
+  - If the `default` case does not apply (i.e., cases cover all execuation paths ), then an
```
Comment (thmiceli) [inline doc/style-guide.md]: _execution_ paths

Comment (ericvw) [inline doc/style-guide.md]: Addressed typo.

**Comment by thmiceli** [comment]
beside the typo LGTM

**Comment by ericvw** [comment]
Updated based on most recent feedback.

**Pull request merged**